docs/source/rules.rst

```diff
@@ -369,6 +372,149 @@ However, the following trigger payload would not match with the ``all`` conditio
+and the action, such as a notification sent to the operator, will be triggered.
+If the second tank had a chemical level of 45, the criteria would resolve to false and no action
+would occur.
+
+Here's an example where the ``condition`` parameter is ``all2any``.
+This will return true in cases where all payload items match any part of the pattern:
+
+.. code-block:: yaml
+
+    ---
+    criteria:
+      trigger.body.data.equipment:
+        type: "search"
+        condition: all2any
+        pattern:
+          item.latitude.value#1:
+            type: "lessthan"
+            pattern: 40
+          item.latitude.value#2:
+            type: "greaterthan"
+            pattern: 50
+          item.longitude.value#1:
+            type: "lessthan"
+            pattern: -100
+          item.longitude.value#2:
+            type: "greaterthan"
+            pattern: -90
+
+Payload:
+
+.. code-block:: json
+
+    {
+    ...
+      "equipment": [
+        {
+          "latitude": {
+            "value": 43
+          }
+          "longitude": {
+            "value": -95
+          }
+        }, {
+          "latitude": {
+            "value": 44
+          }
+          "longitude": {
+            "value": -96
+          }
+        }
+      ]
+    }
+
+In this example all of the equipment coordinates are within the ranges specified by the criteria
+and no action would be taken. If the first equipment latitude were set to 53, there would still
+be no action. If say the second equipment longitude value were set to -106 then the action would
+trigger because ALL of the equipment would be violating at least one of the parts of the pattern.
+This could trigger a notification when the last equipment leaves an area.
+
+
+Single Payload Mode:
+If only a single element is expected in the payload the search parameter can still be used to test the criteria of a rule
+if the payload is a dictionary.
+
+Criteria:
+
+.. code-block:: yaml
+
+    ---
+    criteria:
+      trigger.body.data:
+        type: "search"
+        condition: all2any
+        pattern:
+          item.latitude.value#1:
+            type: "lessthan"
+            pattern: 40
+          item.latitude.value#2:
+            type: "greaterthan"
+            pattern: 50
+          item.longitude.value#1:
+            type: "lessthan"
+            pattern: -100
+          item.longitude.value#2:
+            type: "greaterthan"
+            pattern: -90
+
+Payload:
+
+.. code-block:: json
+
+    {
+      "data": {
+        "latitude": {
+          "value": 43
+        }
+        "longitude": {
+          "value": -95
+        }
+      }
+    }
+
+.. warning::
+
+    When using Single Payload Mode, all2any and any2any has the same result since picking all of one thing is the 
+    same as picking any of one thing.
+
 The search operator is very powerful, but more options for the ``condition`` parameter are
 possible. At this point, only the ``any`` and ``all`` conditions are implemented, but
```

Comment:
Thanks for the contribution (especially the corresponding st2 PR that has the new functionality). Just a minor comment on the documentation: given the addition of new conditions, then this statement below about only "any" and "all" will need updating.

Reply:
updated